app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id_id
+  has_many :comments, foreign_key: :user_id
+  has_many :likes, foreign_key: :user_id
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
```

Comment:
- [ ] Kindly take this function in the post model, a counter should be recorded in the post model.

__HINT:__
```Ruby
def update_post_counter
  author.increment!(:posts_counter)
end
```


![model](https://github.com/toyman640/blog/assets/53124548/55e22b1a-9074-4d8c-833a-802db565bfd1)